src/libs/EmojiUtils.ts

```diff
@@ -1,6 +1,5 @@
 import {getUnixTime} from 'date-fns';
 import {Str} from 'expensify-common';
-import memoize from 'lodash/memoize';
```

Comment:
Let's update our ESLint config to prohibit import of `lodash/memoize` and recommend to instead use our in-house memoize lib.

Reply:
Considering what [You-Dont-Need-Lodash-Underscore](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore) does (and that's already in the project), it would be perfect fit for that task, but it does not support specifying rule for `memoize` method. 

How would you think be the best way to implement it:
- write rule from scratch
- patch [YDNLU](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore)'s [rule.json](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore/blob/master/lib/rules/rules.json) file to accomodate our `memoize` version
- or something different :D

Also, I could not find information if YDNLU provides a way to extend its rules list. It would be the best solution, as patching dependencies should be used as last resort.

Reply:
YDNLU is more for restricting use of lodash and underscore in favor of vanilla JS solutions for which there's a built-in 1:1 replacement. In this case, we've built a custom memoization lib and I don't think it makes sense to try and build this rule into YDNLU.

Instead, this is very easy to accomplish with only a small change to the `.eslintrc.js`:

```diff
diff --git a/.eslintrc.js b/.eslintrc.js
index d8f8c15eb8..46690ee613 100644
--- a/.eslintrc.js
+++ b/.eslintrc.js
@@ -62,6 +62,10 @@ const restrictedImportPaths = [
         importNames: ['Device'],
         message: "Do not import Device directly, it's known to make VSCode's IntelliSense crash. Please import the desired module from `expensify-common/dist/Device` instead.",
     },
+    {
+        name: 'lodash/memoize',
+        message: "Please use '@src/libs/memoize' instead.",
+    },
 ];
 
 const restrictedImportPatterns = [

```

It works well enough for our purposes I think 🙂 

<img width="920" alt="image" src="https://github.com/Expensify/App/assets/47436092/acc3e6d9-00f1-4e58-938c-f649d8cab46a">

Reply:
Done ✅ 

From what I've observed, it does not cover named import from `lodash`:

```ts
// This does not show ESLint error
import { memoize } from 'lodash'
```

Reply:
@kacper-mikolajczak Please check my [comment](https://github.com/Expensify/App/pull/43868#discussion_r1670114599).